demos/asp-net-core/Divergent.Sales.ViewModelComposition/OrdersListViewModelAppender.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Divergent.Sales.ViewModelComposition.Events;
+using ITOps.ViewModelComposition;
+using ITOps.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Divergent.Sales.ViewModelComposition
+{
+    public class OrdersListViewModelAppender : IViewModelAppender
+    {
+        public bool Matches(RouteData routeData, string verb)
+        {
+            /*
+             * matching is a bit weak in this sample, it's designed 
+             * this way to satisfy both the Gateway and the Mvc sample
+             */
+            var controller = (string)routeData.Values["controller"];
+
+            return HttpMethods.IsGet(verb)
+                && controller.ToLowerInvariant() == "orders"
+                && !routeData.Values.ContainsKey("id");
+        }
+
+        public async Task Append(dynamic vm, RouteData routeData, IQueryCollection query)
+        {
+            var pageIndex = (string)query["pageindex"] ?? "0";
+            var pageSize = (string)query["pageSize"] ?? "10";
+
+            var url = $"http://localhost:20295/api/orders?pageSize={pageSize}&pageIndex={pageIndex}";
+            var client = new HttpClient();
+            var response = await client.GetAsync(url).ConfigureAwait(false);
+
+            dynamic[] orders = await response.Content.AsExpandoArrayAsync().ConfigureAwait(false);
+
+            var ordersViewModelDictionary = MapToDictionary(orders);
+
+            await vm.RaiseEventAsync(new OrdersLoaded()
+            {
+                OrdersViewModel = ordersViewModelDictionary
+            }).ConfigureAwait(false);
+
+            vm.Orders = ordersViewModelDictionary.Values.ToArray();
+        }
+
+        IDictionary<dynamic, dynamic> MapToDictionary(dynamic[] orders)
+        {
+            var ordersViewModel = new Dictionary<dynamic, dynamic>();
+
+            foreach (dynamic order in orders)
+            {
+                dynamic vm = new ExpandoObject();
+                vm.OrderNumber = order.Number;
+                vm.OrderItemsCount = order.ItemsCount;
```

Comment:
I'd love to, an initializer would look like:

``` 
dynamic vm = new ExpandoObject()
{
   //set props here
};
```

but compiler complains because `ExpandoObject` doesn't have those properties we're trying to set.